framework/elements/BaseElement.js

```diff
@@ -7,100 +7,114 @@ class BaseElement {
+      const element = await this.getElement()
+      await element.waitForDisplayed({ timeout, reverse, interval })
+    } catch (e) {
+      return false
     }
   }
 
-  async waitForClickable(timeout, reverse = false, interval) {
+  async waitForClickable({ timeout = Timeouts.DEFAULT_WAIT_TIMEOUT, reverse = false, interval = Timeouts.WAIT_FOR_INTERVAL } = {}) {
     try {
-      await this.getElement().waitForClickable({ timeout, reverse, interval: interval ?? Timeouts.WAIT_FOR_INTERVAL })
-    } catch (error) {
-      console.error('Error waiting for element to be clickable:', error)
-      throw error
+      const element = await this.getElement()
+      await element.waitForClickable({ timeout, reverse, interval })
+    } catch (e) {
+      return false
     }
   }
 
-  async waitForEnabled(timeout, reverse = false, interval) {
+  async waitForEnabled({ timeout = Timeouts.DEFAULT_WAIT_TIMEOUT, reverse = false, interval = Timeouts.WAIT_FOR_INTERVAL } = {}) {
     try {
-      await this.getElement().waitForEnabled({ timeout, reverse, interval: interval ?? Timeouts.WAIT_FOR_INTERVAL })
-    } catch (error) {
-      console.error('Error waiting for element to be enabled:', error)
-      throw error
+      const element = await this.getElement()
+      await element.waitForEnabled({ timeout, reverse, interval })
+    } catch (e) {
+      return false
     }
   }
 
-  async isExisting(timeout) {
-    return await this.getElement().isExisting({ timeout })
+  async isExisting({ timeout = Timeouts.DEFAULT_WAIT_TIMEOUT, reverse = false, interval = Timeouts.WAIT_FOR_INTERVAL } = {}) {
+    const element = await this.getElement()
+    return await element.waitForExist({ timeout, reverse, interval })
   }
 
-  async isDisplayed(timeout) {
-    return await this.getElement().isDisplayed({ timeout })
+  async isDisplayed({ timeout = Timeouts.DEFAULT_WAIT_TIMEOUT, reverse = false, interval = Timeouts.WAIT_FOR_INTERVAL } = {}) {
+    const element = await this.getElement()
+    return await element.waitForDisplayed({ timeout, reverse, interval })
   }
 
-  async isClickable(timeout) {
-    return await this.getElement().isClickable({ timeout })
+  async isClickable({ timeout = Timeouts.DEFAULT_WAIT_TIMEOUT, reverse = false, interval = Timeouts.WAIT_FOR_INTERVAL } = {}) {
+    const element = await this.getElement()
+    return await element.waitForClickable({ timeout, reverse, interval })
   }
 
-  async isEnabled(timeout) {
-    return await this.getElement().isEnabled({ timeout })
+  async isEnabled({ timeout = Timeouts.DEFAULT_WAIT_TIMEOUT, reverse = false, interval = Timeouts.WAIT_FOR_INTERVAL } = {}) {
+    const element = await this.getElement()
+    return await element.waitForEnabled({ timeout, reverse, interval })
   }
 
   async getCssProperty(propertyName) {
-    return await this.getElement().getCSSProperty(propertyName).value
+    const element = await this.getElement()
+    return await element.getCSSProperty(propertyName).value
```

Comment:
here it would be better to at first get an object and then return the value or wrap the first part with brackets
`return (await element.getCSSProperty(propertyName)).value`